server/etcdserver/api/snap/snapshotter.go

```diff
@@ -88,7 +89,9 @@ func (s *Snapshotter) save(snapshot *raftpb.Snapshot) error {
 	spath := filepath.Join(s.dir, fname)
 
 	fsyncStart := time.Now()
+	cancel := watchdog.Register("save v2 snapshot")
```

Comment:
I wonder what do you think about pre-registering type of activities that are tracked in the watchdog:  
```
{ at the package level:
  var SAVING_V2_SNAPSHOTS = watchdog.RegisterActivity("saving v2 snapshots");
}```


endActivity := wd.StartActivity(SAVING_V2_SNAPSHOTS)
endActivity()
```
------------
The main benefit is that it drives the API usage to small number of strings rather than patterns like: 

wd.StartActivity(string.format("Writing log entry: %d", entryId));
```
thus the API can be e.g. used for monitoring how long classes of activities take in scope of a watchdog.
i.e. watchdog sees that 17% of wall-time it was in the writing-logs routine. 

It's a stretch and over design. but in theory such registration allows to build also a hierarchy of 'activities/scope'.